load_(un)aligned slightly unergonomic

I was trying to load from memory using an explicit batch type:
```c++
const int8_t* data = ...;
using simd_batch = xsimd::batch<int8_t, 16>;
const simd_batch zero(static_cast<uint8_t>(0));
simd_batch or1 = zero;

// ...
or1 |= xsimd::load_unaligned<simd_batch>(data);
```

but that doesn't work.  I can use the batch constructor like this, but that's not very pretty IMHO:
```c++
or1 |= simd_batch(data, xsimd::unaligned_mode{});
```

(also note the explicit construction for `zero`, which has to a `static_cast` otherwise the constructor call is ambiguous)

Are you sure? I get the following error:
```
../src/arrow/util/utf8.h: In function 'bool arrow::util::ValidateAsciiSimd(const uint8_t*, int64_t)':
../src/arrow/util/utf8.h:257:78: error: cannot call member function 'xsimd::batch<T, N>& xsimd::sse_int_batch<T, N>::load_unaligned(const T*) [with T = signed char; long unsigned int N = 16]' without object
  257 |       or1 |= simd_batch::load_unaligned(reinterpret_cast<const int8_t*>(data));
      |                                                                              ^
```

According to the docs, it's a non-static member function that loads into `*this`.

--- a/include/xsimd/types/xsimd_base.hpp
+++ b/include/xsimd/types/xsimd_base.hpp
@@ -153,6 +153,14 @@ namespace xsimd
         using reverse_iterator = std::reverse_iterator<iterator>;
         using const_reverse_iterator = std::reverse_iterator<const_iterator>;
 
+        static X broadcast(value_type v);
+
+        template <class T>
+        static X from_unaligned(T* src);
+
+        template <class T>
+        static X from_aligned(T* src);
+
         X& operator+=(const X& rhs);
         X& operator+=(const value_type& rhs);
 
@@ -676,6 +684,52 @@ namespace xsimd
     {
     }
 
+    /**
+     * @name Static builders
+     */
+    //@{
+    /**
+     * Creates a batch from the single value \c v.
+     * @param v the value used to initialize the batch
+     * @return a new batch instance
+     */
+    template <class X>
+    inline X simd_batch<X>::broadcast(value_type v)
+    {
+        return X(v);
+    }
+
+    /**
+     * Creates a batch from the buffer \c src. The
+     * memory does not need to be aligned.
+     * @param src the memory buffer to read
+     * @return a new batch instance
+     */
+    template <class X>
+    template <class T>
+    inline X simd_batch<X>::from_unaligned(T* src)
+    {
+        X res;
+        res.load_unaligned(src);
+        return res;
+    }
+
+    /**
+     * Creates a batch from the buffer \c src. The
+     * memory needs to be aligned.
+     * @param src the memory buffer to read
+     * @return a new batch instance
+     */
+    template <class X>
+    template <class T>
+    inline X simd_batch<X>::from_aligned(T* src)
+    {
+        X res;
+        res.load_aligned(src);
+        return res;
+    }
+    //@}
+    
     /**
      * @name Arithmetic computed assignment
      */

--- a/test/test_batch.cpp
+++ b/test/test_batch.cpp
@@ -62,6 +62,30 @@ protected:
         EXPECT_EQ(b1, lhs) << print_function_name("batch(value_type*)");
     }
 
+    void test_static_builders() const
+    {
+        {
+            array_type expected;
+            std::fill(expected.begin(), expected.end(), value_type(2));
+
+            auto res = batch_type::broadcast(value_type(2));
+            EXPECT_EQ(res, expected) << print_function_name("batch::broadcast");
+        }
+        {
+            array_type res;
+            auto b = batch_type::from_unaligned(lhs.data());
+            b.store_unaligned(res.data());
+            EXPECT_EQ(res, lhs) << print_function_name("batch::from_unaligned");
+        }
+        {
+            alignas(XSIMD_DEFAULT_ALIGNMENT) array_type arhs(this->rhs);
+            alignas(XSIMD_DEFAULT_ALIGNMENT) array_type ares;
+            auto b = batch_type::from_aligned(arhs.data());
+            b.store_aligned(ares.data());
+            EXPECT_EQ(ares, rhs) << print_function_name("batch::from_aligned");
+        }
+    }
+
     void test_access_operator() const
     {
         batch_type res = batch_lhs();
@@ -547,6 +571,11 @@ TYPED_TEST(batch_test, load_store)
 TYPED_TEST(batch_test, constructors)
 {
     this->test_constructors();
+}
+
+TYPED_TEST(batch_test, static_builders)
+{
+    this->test_static_builders();
 }
 
 TYPED_TEST(batch_test, access_operator)

